src/Sylius/Bundle/CoreBundle/Checkout/Step/FinalizeStep.php

```diff
@@ -89,5 +89,8 @@ protected function completeOrder(OrderInterface $order)
 
         $this->dispatchCheckoutEvent(SyliusCheckoutEvents::FINALIZE_COMPLETE, $order);
         $this->dispatchCheckoutEvent(SyliusOrderEvents::POST_CREATE, $order);
+
+        $cartProvider = $this->getCartProvider();
+        $cartProvider->abandonCart();
```

Comment:
It's confusing to dispatch abandon event... but I suppose we can change that after we define the final order states ([see](https://github.com/Sylius/Sylius/issues/4060) pending discussion).